src/views/UserView/UserView.js

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import Logger from "js-logger";
+import { fetchTrendingVideo, getUserDetails } from "../../services/videoApi";
+
+import UserProfile from "../../components/UserProfile";
+import UserVideoList from "../../components/UserVideoList";
+import UserVideoItem from "../../components/UserVideoItem";
+import Loading from "../../components/Loading";
+
+const UserView = () => {
+  const { name } = useParams();
+  const [userDetails, setUserDetails] = useState({
+    data: {},
+    error: "",
+    loading: true,
+  });
+  const [trendingVideos, setTrendingVideos] = useState({
+    data: [],
+    error: "",
+    loading: true,
+  });
+
+  useEffect(() => {
+    async function innerFunction() {
+      try {
+        const response = await fetchTrendingVideo();
+        setTrendingVideos(() => ({
+          data: response.data,
+          error: "",
+          loading: false,
+        }));
+      } catch (error) {
+        setTrendingVideos(() => ({
+          data: [],
+          error: error.message,
+          loading: false,
+        }));
+      }
+    }
+
+    Logger.trace(innerFunction());
+  }, []);
```

Comment:
Це і отримання **UserDetails** можна винести в окремий кастомний хук

Reply:
Що ти маєш на у вазі? Який кастомний хук? Об'єднати два запити у один? Можна приклад?

Reply:
Хук - це ж простими словами функція, яка виконує якусь логіку і щось нам повертає, просто в назві префікс **use**. Наприклад, створюємо файл useFetchTrendVideo.js, в ньому називаємо так і сам хук. Цю логіку яку я виділив виносимо в цей файл.
```suggestion
export const useFetchTrendVideo = () => {

	const [trendingVideos, setTrendingVideos] = useState({
		data: [],
		error: "",
		loading: true,
	})

	useEffect(() => {
		async function innerFunction() {
			try {
				const response = await fetchTrendingVideo()
				setTrendingVideos(() => ({
					data: response.data,
					error: "",
					loading: false,
				}))
			} catch (error) {
				setTrendingVideos(() => ({
					data: [],
					error: error.message,
					loading: false,
				}))
			}
		}

		Logger.trace(innerFunction())
	}, [])

	return trendingVideos
}
```

Reply:
Відповідно потрібно зробити всі необхідні імпорти. Далі у файлі **UserView.js** ти використовуєш цей хук як звичайний `useState`.

`const trendingVideos = useFetchTrendVideo() `

Але ти можеш піти ще далі та спробувати всі axios запити об'єднати в один хук.

Reply:
Готово